.github/actions/ansible_azure_test_provider/create_session.py

```
@@ -0,0 +1,67 @@
+#!/usr/bin/python
+"""Script to request new azure session using ansible_core_ci_key."""
+
+import json
+import os
+import secrets
+import sys
+
+from pathlib import PosixPath
+
+import requests
+
+
+def main() -> None:
+    """Request new azure session credentials.
+
+    :raises ValueError: when ANSIBLE_CORE_CI_KEY environment variable is missing or empty
+    """
+    ansible_core_ci_key = os.environ.get("ANSIBLE_CORE_CI_KEY") or ""
```

Why not just use `os.environ["ANSIBLE_CORE_CI_KEY"]` and let it raise the exception if not present? Or if you want to supply a different message, handle that exception here?

---

updated, thanks